plugins/logs/plugin_test.go

```diff
@@ -1561,6 +1616,22 @@ func TestPluginMasking(t *testing.T) {
 			},
 			expPrinted: []string{"Erasing /input/password"},
 		},
+		{
+			note: "simple upsert on ndbcache",
+			rawPolicy: []byte(`
+				package system.log
+				mask[{"op": "upsert", "path": "/ndbcache/rand.intn", "value": x}] {
+					x := "**REDACTED**"
+				}`),
+			expMasked: []string{"/ndbcache/rand.intn"},
+			ndbcache: map[string]interface{}{
+				// Simulate rand.intn("z", 15) call, with output of 7.
+				"rand.intn": ast.NewObject([2]*ast.Term{ast.StringTerm("[\"z\",15]"), ast.IntNumberTerm(7)}),
+			},
+			ndbc_expected: map[string]interface{}{
+				"rand.intn": "**REDACTED**",
```

Comment:
Can we add one more test case where there are multiple values for a nd builtin?

Reply:
Yep! Added a more complex case where one of the entries for a builtin is modified in-place.